includes/class-flizpay-gateway.php

```diff
@@ -934,14 +934,35 @@ public function create_transaction($order, $source = 'plugin')
                 'firstName' => $order->get_billing_first_name(),
                 'lastName' => $order->get_billing_last_name()
             );
+            $products = array();
+            foreach ( $order->get_items() as $item_id => $item ) {
+                $product = $item->get_product();
+                if ( !$product ) {
+                    continue; 
+                }
+  
+                $name    = $product->get_name();
+                $amount  = $item->get_quantity();
+                $price   = $product->get_price(); // or use $item->get_subtotal() depending on how you want to handle pricing

```

Comment:
```suggestion
                $price   = $product->get_subtotal(); 
```

So the price comes with any sales promotion or discount